src/services/LocationService.ts

```diff
@@ -36,18 +53,51 @@ class LocationService {
    * Retrieves the user's current location using navigator.geolocation.
    * Times out and calls the error callback after 3s.
    */
-  public getCurrentLocation(): Promise<GeolocationPosition> {
+  public async getCurrentLocation(): Promise<GeolocationPosition> {
     return new Promise((resolve, reject) => {
       if (navigator.geolocation) {
         const onSuccess = (position: GeolocationPosition) => resolve(position);
         const onError = (error: GeolocationPositionError) => reject(error);
-        const options = { timeout: 3000 };
-        navigator.geolocation.getCurrentPosition(onSuccess, onError, options);
+        navigator.geolocation.getCurrentPosition(onSuccess, onError, { timeout: 3000 });
+      } else {
+        resolve(this.defaultLocation);
+      }
+    });
+  }
+}
+
+class LatLngService extends LocationService<google.maps.LatLng> {
+  /**
+   * Returns the default location as a google.maps.LatLng object.
+   */
+  public get defaultLocation(): google.maps.LatLng {
+    const { latitude, longitude } = this.defaultCoordinates;
+    return new google.maps.LatLng(latitude, longitude);
+  }
+
+  /**
+   * Retrieves the user's current location using navigator.geolocation.
+   * Times out and calls the error callback after 3s.
+   */
+  public async getCurrentLocation(): Promise<google.maps.LatLng> {
+    return new Promise((resolve, reject) => {
+      if (navigator.geolocation) {
+        const onSuccess = (position: GeolocationPosition) => {
+          const {
+            coords: { latitude, longitude },
+          } = position;
+          return resolve(new google.maps.LatLng(latitude, longitude));
+        };
+        const onError = (error: GeolocationPositionError) => reject(error);
+        navigator.geolocation.getCurrentPosition(onSuccess, onError, { timeout: 3000 });
       } else {
         resolve(this.defaultLocation);
       }
     });
   }
 }
 
-export default new LocationService();
+export default {
+  GeolocationPosition: new GeolocationPositionService(),
```

Comment:
Do we have two different location services in case one of the two doesn't work?

Reply:
Ahh the two location services are mainly to provide location info in different types, namely `GeolocationPosition` and `LatLng`. Some places we need `LatLng` objects (I think anything to do with Google APIs) whereas everywhere else we don't really need to convert the GeolocationPosition to LatLng types!